client/state/signup/dependency-store/reducers.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SIGNUP_DEPENDENCY_STORE_RESET,
+	SIGNUP_DEPENDENCY_STORE_UPDATE_STATE,
+} from 'state/action-types';
+
+export function storeState( state = null, action ) {
+	switch ( action.type ) {
+		case SIGNUP_DEPENDENCY_STORE_UPDATE_STATE:
+			return Object.assign( {}, state, action.data );
```

Comment:
I would suggest adding a few more actions here, which we can explicitly dispatch when a user clicks on an answer during signup.

```javascript
export const signupSurvey = createReducer( {}, {
    [ SIGNUP_SURVEY_SITE_SET ]: ( state, action ) => { ...state, site: action.value },
    [ SIGNUP_DESIGN_TYPE_SET ]: ( state, action ) => { ...state, designType: action.value } //... and so on
} );
```

We may also want to store which step we're on. Which might look similar to:
```javascript
export const currentStep = createReducer( 'SITE_TYPE', {
	[ SIGNUP_SURVEY_SITE_SET ]: () => 'DESIGN',
        [ SIGNUP_DESIGN_TYPE_SET ]: () => 'DOMAIN' //... and so on
} );

export default combineReducers( {
    signupSurvey,
    currentStep
} );
```